src/__tests__/tuf/trustroot.test.ts

```diff
@@ -0,0 +1,134 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { Updater } from 'tuf-js';
+import { TrustedRootError } from '../../error';
+import { TrustedRootFetcher } from '../../tuf/trustroot';
+
+describe('TrustedRootFetcher', () => {
+  const publicKey = `-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEGg6Hjxt2UNiJ1kwwq5XQIIwMZnJf
+VQ3bF01uZKteMdcV/3qhCmWOecoxRqwrbYTshGg9NyXcBbve6zKwZVTLeg==
+-----END PUBLIC KEY-----`;
+
+  // Set-up a directory to use as our TUF cache and write a public key to it.
+  const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'tuf-cache-test-'));
+  const targetFile = path.join(tempDir, 'key.pub');
+  fs.writeFileSync(targetFile, publicKey);
+
+  afterAll(() => {
+    fs.rmSync(tempDir, { recursive: true });
+  });
+
+  const caCertTarget = {
+    custom: {
+      sigstore: {
+        uri: 'https://fulcio.sigstore.dev',
+        usage: 'Fulcio',
+        status: 'active',
+      },
+    },
+  };
+
+  const tlogKeyTarget = {
+    custom: {
+      sigstore: {
+        uri: 'https://rekor.sigstore.dev',
+        usage: 'Rekor',
+        status: 'active',
+      },
+    },
+  };
+
+  const ctlogKeyTarget = {
+    custom: {
+      sigstore: {
+        uri: 'https://ctfe.sigstore.dev',
+        usage: 'CTFE',
+        status: 'active',
+      },
+    },
+  };
+
+  describe('getTrustedRoot', () => {
+    describe('when the targets can be retrieved', () => {
+      // Dummy TUF Updater
+      const tuf = {
+        refresh: jest.fn(),
+        findCachedTarget: jest.fn().mockReturnValue(targetFile),
+        trustedSet: {
+          targets: {
+            signed: { targets: [caCertTarget, tlogKeyTarget, ctlogKeyTarget] },
+          },
+        },
+      };
+
+      const subject = new TrustedRootFetcher(tuf as unknown as Updater);
+
+      it('assembles and returns the TrustedRoot', async () => {
+        const root = await subject.getTrustedRoot();
+
+        expect(root).toBeDefined();
+        expect(root.certificateAuthorities).toHaveLength(1);
+        expect(root.tlogs).toHaveLength(1);
+        expect(root.ctlogs).toHaveLength(1);
+        expect(root.timestampAuthorities).toHaveLength(0);
+
+        const ca = root.certificateAuthorities[0];
+        expect(ca.uri).toEqual('');
```

Comment:
Might be misunderstanding something here, but I would expect the `uri` to be `https://fulcio.sigstore.dev`?

Reply:
Since I'm constructing a single certificate authority entry from multiple 'Fulcio' targets, I didn't bother to collect/combine the URIs to populate this field (also, it's not being used anywhere else in the system at the moment).

This will all be much cleaner once we start consuming the actual `TrustedRoot`.